### PR TITLE
Add support for Microsoft AD FS

### LINKED ISSUE #352: unable to use with Microsoft AD FS server
### Jenkins and plugins versions report

`mvn hpi:run` from master (cddbd8ae4694860c0115cb784f87eaa3f32df2cf)

<details>
  <summary>Environment</summary>

Microsoft AD FS server (2022)  https://learn.microsoft.com/en-us/windows-server/identity/ad-fs/operations/set-up-an-ad-fs-lab-environment#install-the-ad-fs-server-role

  
  ```text
nap
  ```

</details>

For whatever reason and despite advertising differently the AD FS server is usign HTTP POST for the return URL.

https://learn.microsoft.com/en-us/windows-server/identity/ad-fs/overview/ad-fs-openid-connect-oauth-flows-scenarios

>  response_mode optional Specifies the method that should be used to send the resulting token back to your app. Defaults to fragment.

This causes issues as the crumb issuer intercepts the call finds there is no crumb and says go away.  

### What Operating System are you using (both controller, and any agents involved in the problem)?

Windows.

### Reproduction steps

1. install an AD FS server as per MSFT "ADSL Lab enironment"
2. create a client for jenkins
3. mvn hpi:run and configure oic plugin
4. attempt to login

### Expected Results

login works

### Actual Results

login fails as the reirect is a crumb

### Anything else?

specifying `response_mode` as `fragement` or `query` which should be the default should force this even if it is supposed to be the default.

### Are you interested in contributing a fix?

_No response_

### PR DESCRIPTION
**Pull request by jtnord**
Enable the plugin to work with [AD FS](https://learn.microsoft.com/en-us/windows-server/identity/ad-fs/ad-fs-overview) which sends the user back to `securityRealm/finishLogin` with a HTTP Post.

Adds a Crumb exclusion extension for this endpoint.
Extract any POST data provided into a fake URL so it can be fed to the google code in a way that it can work with
Adds some rudimentary docs

fixes #352

### Testing done

* setup a local LAB ADFS (OMG!) and configured the plugin to use it.
* validated I can login and obtain groups.
* unit test for the crumb exclusion

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: docs/configuration/ADFS.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+It also supports PKCE verification for additional security.
+
+### User information
+
+Without any extra claims, the user field should be set to `upn`
+
+
+[1]: https://learn.microsoft.com/en-us/windows-server/identity/ad-fs/ad-fs-overview
+[2]: https://stackoverflow.com/questions/55494354/user-groups-as-claims-through-openid-connect-over-adfs/55570286#55570286
+[3]:
```
Comment (jtnord) [inline docs/configuration/ADFS.md:26]:
```suggestion
[3]: https://community.ibm.com/community/user/security/blogs/laurent-lapiquionne1/2020/07/21/how-to-configure-igi-service-center-to-authent
```

File: pom.xml [hunk 2]
```diff
@@ -129,6 +129,36 @@
   </pluginRepositories>
 
   <build>
+    <pluginManagement>
```
Comment (jtnord) [inline pom.xml:132]: unrelated to the change, but due to eclipse-m2e/m2e-core/issues/1291 would cause an error in the project in eclipse.

**Comment by jtnord** [comment]
@michael-doubez the metrics for code coverage appear broken.

https://app.codecov.io/gh/jenkinsci/oic-auth-plugin/pull/368?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=jenkinsci#7dda26682abc4c20fa99de9ee4c7c0aa-L1423 is showing -0.75% coverage (`[src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java](https://app.codecov.io/gh/jenkinsci/oic-auth-plugin/pull/368/blob/src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java)`) yet this class in unchanged nor do any changes in test or production code affect it..  

The[ base used](https://github.com/jenkinsci/oic-auth-plugin/commit/b7205f1c7e76f90f818e4a7716c3009d5fdbfd75) for the diff is 11 commits behind master, yet the base for this PR is https://github.com/jtnord/oic-auth-plugin/commit/3df533875ca18afb1e96b6efbb1058e3b0751c6a which is the same as `jenkinsci/master` at the time of writing this.

image

**Comment by jtnord** [comment]
code coverage [upload is failing](https://github.com/jenkinsci/oic-auth-plugin/actions/runs/10365897501/job/28693988638#step:5:34) on the master branch

```
Error: Codecov token not found. Please provide Codecov token with -t flag.
Warning: Codecov: Failed to properly create commit: The process '/home/runner/work/_actions/codecov/codecov-action/v4/dist/codecov' failed with exit code 1
```

**Review by jmdesprez — APPROVED** [review]

**Review by michael-doubez — APPROVED** [review]
LGTM

**Pull request merged**